### ISSUE TITLE
Silhouette inconsistency

### ISSUE THREAD

**Issue by AlexHermansson** [kno10/python-kmedoids#23]
There is an inconsistency with the method to compute sample and average silhouette scores with `kmedoids.silhouette()` compared to sklearn's method, they give very different results. Is there a reason for this? Both methods refer to the same paper where the silhouette score was introduced.
Thank you.

**Comment by kno10** [comment]
Please provide an example where this happens.

**Comment by kno10** [comment]
Ok, I can reproduce this. @lale1009 can you debug and add a unit test?

**Comment by AlexHermansson** [comment]
Hi, even after the fix the results are not the same. 
Here's an example:

```
import kmedoids
import numpy as np
from sklearn.metrics import silhouette_samples

dist_matrix = np.array([[0.0, 0.1, 0.3],
                        [0.1, 0.0, 0.2],
 		        [0.3, 0.2, 0.0]])

labels = np.array([0, 0, 1])

silh_samples = silhouette_samples(dist_matrix, labels) # sklearn
_, silh_samples2 = kmedoids.silhouette(dist_matrix, labels, samples=True) # kmedoids

print(silh_samples)
print(silh_samples2)
```

Output:

```
[0.60264029 0.5        0.        ]
[0.66666667 0.5        0.        ]

```

**Comment by lale1009** [comment]
Hi @AlexHermansson,
Please try: silh_samples = silhouette_samples(dist_matrix, labels, metric = 'precomputed')

**Comment by AlexHermansson** [comment]
Right, I was too quick writing that little test.. Sorry. Thanks a lot for fixing this issue so fast, great work with this library. I really appreciate it!
